coralproject/cay
# Oidc hookup

## What does this PR do?

Opening this PR so I can get some feedback on how I should architect things. What I've done so far is implement the `redux-oidc` middleware (which is really just a wrapper for the `oidc-client-js` library. We're waiting on Shelf to have a deploy that will handle these auth headers, so some of this is a moot point.

TODO
- [x] build a modal that prompts the user to log back in (generating a new token) after 1 hour
- [x] figure out how/if the middleware checks that the token is expired
- [x] decide on best strategy for passing the `Authorization` header. I've taken a stab at this in `FormActions.js`
- [x] when is the token checked? every action?
- [x] add some auth error UI feedback elements
- [x] store `id_token` in `localStorage` so login is not lost on refresh
## How do I test this PR?
- run the [coral-auth](https://github.com/coralproject/coral-auth) docker image
- generate your auth keys in coral-auth
- update the `.env` file with your settings. check the README in coral-auth
- create a user for yourself on the cli
- `npm install` in cay
- install ngrok from homebrew
- point ngrok at the coral-auth instance with `$ ngrok http 5000`
- copy/paste the ngrok https url into the `authAuthority` field in `/public/config.json` with `/connect`. e.g. `"authAuthority": "https:948u983.ngrok.io/connect"`
- copy your client id from the `.env` file into an `authClientId` field in `/public/config.json`
- go to `/login` and try to login

@coralproject/frontend
